### DISCUSSION TITLE
Setting the "draggable" property to "false"

### DISCUSSION THREAD

**Discussion by cerk** [rabbibotton/clog#297] [General]
In case this helps anyone else because it was not obvious to me.

I had some code where I wanted an html div to have the "draggable" attribute set to "false." Here was my first attempt:
```
(defun clog-draggable-test ()
  (defun on-new-window (body)
    (let* ((img-div (create-img body :url-src "image.png")))
      (setf (draggablep img-div) nil)))
  (initialize #'on-new-window :static-root "/home/usr/images/")
  (open-browser))
```
After running this and inspecting the img tag in the browser, I saw:
```
<img src="image.png" alt="" id="CLOG8187" draggable="true">
```
instead of what I wanted which was:
```
<img src="image.png" alt="" id="CLOG8187" draggable="false">
```
I searched for similar problems and found these links:
[https://github.com/sveltejs/svelte/issues/2935](https://github.com/sveltejs/svelte/issues/2935)
[https://github.com/lit/lit-element/issues/565](https://github.com/lit/lit-element/issues/565)
[MDN Web Docs draggable definition](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/draggable) (see the warning section)

Which led me to this code:
```
(defun clog-draggable-test ()
  (defun on-new-window (body)
    (let* ((img-div (create-img body :url-src "image.png")))
      (setf (attribute img-div "draggable") "false")))
  (initialize #'on-new-window :static-root "/home/usr/images/")
  (open-browser))
```

**Comment by aykaramba** [comment]
Huh.  Interesting because setting visibility is done with NIL as per:  (setf (hiddenp container-of-some-sort) nil)

Thanks for that, I will keep that in mind.

**Comment by aykaramba** [comment]
As a side note:

1) Thanks for helping me understand the drag and drop behaviour.

2) I have code for uploading images using a regular form.  I will upload that in a bit for reference.  I suspect you won't need it but it would be a good item for reference for other noobs,.

3) So the drag / drop + upload stuff would lead us to start thinking about what drag / drop ui would look like for uploading multiple files asynchronously using some javascript interface.   Yes, that would be uploading from local to cloud, however, once uploaded the drag / drop stuff could allow for manual sorting as necessary which could make an interesting set of widgets.

Thanks for pushing all of this forward.
